pkg/controller/common/resource_reconciler_test.go

```diff
@@ -2,20 +2,34 @@ package common
 
 import (
 	"context"
+	"os"
+	"testing"
+	"time"
+
 	"github.com/Azure/aks-app-routing-operator/pkg/controller/metrics"
 	"github.com/Azure/aks-app-routing-operator/pkg/controller/testutils"
-	"testing"
 
 	"github.com/go-logr/logr"
 	"github.com/stretchr/testify/require"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 )
 
+var restConfig *rest.Config
+
+func TestMain(m *testing.M) {
+	restConfig, _ = testutils.StartTestingEnv()
+
+	exitCode := m.Run()
```

Comment:
moving the cleanup to a defer right under the start might make sense.

```Go
restCfg, err := testutils.StartTestingEnv()
if err != nil {
   panic(err)
}
defer testutils.CleanupTestingEnv()

os.Exist(m.Run())
```

Reply:
Added panic and defer